numba/cloudpickle/cloudpickle.py

```diff
@@ -203,6 +203,10 @@ def _lookup_module_and_qualname(obj, name=None):
 
     module_name = _whichmodule(obj, name)
 
+    if module_name not in sys.modules:
+        del obj.__module__
```

Comment:
Why is it necessary to do this?

(To my surprise, I didn't know that deleting `__module__` is the same as setting it to `None`.)

Reply:
If we don't delete the module from the object then it will try to import it the module at line 97 in fundesc.py. It will then raise the module import error and fail. We could maybe deal with it at that point, but I'm not very familiar with the code and so tried to make it as similar as possible to the old serialisation code.